### DISCUSSION TITLE
Room/Space's location : find space and room nearby me.

### DISCUSSION THREAD

**Discussion by Thatoo** [element-hq/element-meta#484] [Ideas]
#### What would you like to do?
Be able to set a location to a specific room.
A local association has its own public room to be contacted by people from its "territory".
It sets a location in settings of their room.
People can browse rooms by location thanks to a browsing address system and a mapping system (if they want to browse in a location they might go traveling or move in or...), and of course see all public room (which have set their location) around themselves (thanks to user's location tool).
#### Why would you like to do it?
People might find interesting rooms talking about interesting subject around where they live. They might get in touch and meet new people around themselves to contribute locally.
#### How would you like to achieve it?
Location system is already in Element so I guess the most difficult part is already done.
It just requires to allow setting a location in room settings (address and GPS coordinate).
Then, it needs to add a location filter (by address or by X km around the user location and by map browsing) in the public room browsing system of Element.
It also need a location filter in the room browsing system of a particular space the user is member of.

**Comment by Thatoo** [comment]
@SimonBrandner sorry to ask but could you tell me why you add the uncommon flag? It is not because other chat app don't offer this option that it isn't useful. Why would we not facilitate matrix users to find groups (association...) using matrix to communicate who are nearby IRL?
For example, someone living in Lorient or Lanester in France could find #accueil:defis.info a welcoming room of the association Defis who is helping people to learn basic computing using free software only...
If the user don't know yet Defis, the user can't find easily this room. But if location was set in the setting of this room and the user could get a list of room nearby oneself, the user might find it easily and come join into it and, maybe, become a volunteer some day and maintain the synapse server :-)

Hoping an example would help to understand the usage of it.
Of course location setting of room is optional.

**Comment by Thatoo** [comment]
Related to https://github.com/vector-im/element-meta/discussions/1341

**Comment by t3chguy** [comment]
Closing in favour of https://github.com/vector-im/element-meta/discussions/1341
